cmd/registry-experimental/cmd/compute/descriptor.go

```diff
@@ -128,3 +140,59 @@ func (task *computeDescriptorTask) Run(ctx context.Context) error {
 	}
 	return core.SetArtifact(ctx, task.client, artifact)
 }
+
+// descriptorFromZippedProtos runs protoc and returns the results.
+func descriptorFromZippedProtos(ctx context.Context, name string, b []byte) (*descriptorpb.FileDescriptorSet, error) {
+	// create a tmp directory
+	root, err := ioutil.TempDir("", "registry-protos-")
+	if err != nil {
+		return nil, err
+	}
+	// whenever we finish, delete the tmp directory
+	defer os.RemoveAll(root)
+	// unzip the protos to the temp directory
+	_, err = core.UnzipArchiveToPath(b, root+"/protos")
+	if err != nil {
+		return nil, err
+	}
+	return generateDescriptorForDirectory(ctx, name, root)
+}
+
+func generateDescriptorForDirectory(ctx context.Context, name string, root string) (*descriptorpb.FileDescriptorSet, error) {
+	// run protoc on all of the protos in the main directory
+	protos := []string{}
+	err := filepath.Walk(root+"/protos",
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if strings.HasSuffix(path, ".proto") {
+				protos = append(protos, strings.TrimPrefix(path, root+"/"))
+			}
+			return nil
+		})
+	if err != nil {
+		return nil, err
+	}
+	args := []string{}
+	args = append(args, protos...)
+	args = append(args, "--proto_path=protos")
+	args = append(args, "--descriptor_set_out=proto.pb")
+	cmd := exec.Command("protoc", args...)
+	cmd.Dir = root
+	log.FromContext(ctx).Debugf("Running %+v", cmd)
+	data, err := cmd.CombinedOutput()
+	if err != nil {
+		log.FromContext(ctx).Errorf("%+v", err)
```

Comment:
Line 125 already logs this error, although at a different severity level. IMO we should remove this one and let it log at the higher level.

Reply:
Thanks, I'll fix that.

Reply:
Done.